test/pylib/cpp/boost/prepare_instance.py

```diff
@@ -0,0 +1,116 @@
+#
+# Copyright (C) 2024-present ScyllaDB
+#
+# SPDX-License-Identifier: AGPL-3.0-or-later
+#
+from __future__ import annotations
+
+import asyncio
+import logging
+import os
+import time
+from pathlib import Path
+
+from test.pylib.host_registry import HostRegistry
+from test.pylib.minio_server import MinioServer
+from test.pylib.s3_server_mock import MockS3Server
+from test.pylib.util import LogPrefixAdapter
+
+
+class PrepareChildProcessEnv:
+    """
+    Class responsible to get environment variables from the main thread through the shared file and set them for the process
+    """
+    def __init__(self, env_file: Path):
+        self.env_file = env_file
+
+    def prepare(self) -> None:
+        """
+        Read the environment variables for S3 and MinIO from the file and set them for the process.
+        """
+        timeout = 10
+        start_time = time.time()
+        while True:
+            if os.path.exists(self.env_file):
+                with open(self.env_file, 'r') as file:
+                    for line in file.readlines():
+                        key, value = line.strip().split('=', 1)
+                        os.environ[key] = value
+                break
+
+            if time.time() - start_time > timeout:
+                raise TimeoutError(f"Timeout waiting for file {self.env_file}")
+            # Sleep needed to wait when the controller will create a file with environment variables.
+            # Without sleep checking of the file existence will be too fast,
+            # so it will finish before the file is created
+            time.sleep(0.5)
+
+    def cleanup(self) -> None:
+        """
+        Fake method to have the same interfaces with Controller class.
+        """
+        pass
+
+
+class PrepareMainProcessEnv:
+    """
+    A class responsible for starting additional services needed by tests.
+
+    It starts up a Minio server and an S3 mock server.
+    The environment settings are saved to a file for later consumption by child processes.
+    """
+    def __init__(self, temp_dir: Path, modes: list[str], env_file: Path):
+        self.temp_dir = temp_dir
+        pytest_dirs = [self.temp_dir / mode / 'pytest' for mode in modes]
+        for directory in [self.temp_dir, *pytest_dirs]:
+            if not directory.exists():
+                directory.mkdir()
+        self.env_file = env_file
+        hosts = HostRegistry()
+        self.loop = asyncio.new_event_loop()
+        address = self.loop.run_until_complete(hosts.lease_host())
+        self.ms = MinioServer(self.temp_dir, '127.0.0.1', LogPrefixAdapter(logging.getLogger('minio'), {'prefix': 'minio'}))
```

Comment:
Why do you use 127.0.0.1 and not "address" here? Is this a mistake?

Reply:
Also nitpick - maybe "ms" is too short of a name? Why not call it "minio"? It confused me later in the code.